.github/workflows/main.yaml

```diff
@@ -194,6 +194,49 @@ jobs:
           cd ../..
           Rscript -e "devtools::test(reporter = c(\"summary\", \"fail\"))"
 
+  test_rust_client:
+    needs: [build_test_models]
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+      fail-fast: false
+    steps:
+      - name: Check out github
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Restore Stan
+        uses: actions/cache@v3
+        with:
+          path: ./stan/
+          key: ${{ runner.os }}-stan-${{ hashFiles('stan/src/stan/version.hpp') }}-v${{ env.CACHE_VERSION }}
+
+      - name: Restore built models
+        uses: actions/cache@v3
+        id: test-models
+        with:
+          path: ./test_models/
+          key: ${{ hashFiles('**/*.stan', 'src/*', 'stan/src/stan/version.hpp') }}-${{ matrix.os }}-v${{ env.CACHE_VERSION }}
+
+      - name: Install LLVM and Clang  # Required for bindgen
+        uses: KyleMayes/install-llvm-action@v1
+        with:
+          version: "15.0"
+          directory: ${{ runner.temp }}/llvm
```

Comment:
I noticed that the `bindgen` crate itself doesn't seem to need to manually install LLVM. They use `actions-rs/toolchain` instead of the action from `dtolnay`, is it possible that does it for us?

Reply:
I think they just don't test on windows, and I don't see anything in actions-rs that would get us libclang...

Reply:
This part of the github action is installing llvm/clang on Ubuntu and OSX, not Windows. Is it still necessary?